### PR TITLE
Milestone 3: CRUD functions for weapons

### PR DESCRIPTION
**Pull request by willkotheimer**
Star Destroyer website needs to have a weapons dashboard. We did it!

## Description

It includes Create, Read, Update and delete functions for logged in users, and READ only functions for logged out users.

## Related Issue

[https://github.com/nss-evening-cohort-13/star-destroyer-dashboard-star-destroyer-dashboard/issues/37](https://github.com/nss-evening-cohort-13/star-destroyer-dashboard-star-destroyer-dashboard/issues/37)

## Motivation and Context

This was created as part of the submission for the nutshell project -- Star Destroyer. The purpose is to use firebase and axios for auth and CRUD purposes.

## How Can This Be Tested?

git pull origin development
git checkout development
npm start

###

URL [https://star-destroyer-a0766.web.app](https://star-destroyer-a0766.web.app)

## Screenshots (if appropriate):

NPM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/javascripts/components/cards/weaponsCards.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  `;
+  }
+
+  const domString = $('#app').append(`
+    <div class="weapons" id="weapons-${object.firebaseKey}">
+  <div class="card" style="width: 18rem;">
+  <img src="${object.image}" class="card-img-top" alt="...">
+  <div class="card-body">
+    <h5 class="card-title" id="name">${object.name}</h5>
+    ${buttons}
```
Comment (drteresavasquez) [inline src/javascripts/components/cards/weaponsCards.js:19]: LOVE!!!

File: src/javascripts/components/forms/updateWeaponsForm.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      </div>
+      <div class="form-group"
+        <label for="image">Image Url</label>
+        <input type="text" class="form-control" value="${weaponsObj.image}" id="p-image" placeholder="Enter image Url">
+      </div>
+      <button id="update-weapons-btn" type="submit" class="btn btn-info"><i class="fas fa-plus-circle"></i> Update Weapons</button>
+    </form>`);
+
+  weaponsData.getAllWeapons().then((response) => {
+    response.forEach((item) => {
```
Comment (drteresavasquez) [inline src/javascripts/components/forms/updateWeaponsForm.js]: Where is this being used? I don't see a select menu. Remove this if it is not in use

File: src/javascripts/components/views/weaponsView.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    if (response.length) {
+      response.forEach((item) => {
+        $('#app').append(card.weaponsCard(item));
+      });
+    } else {
+      $('#app').append('<h2> NO CARDS!</h2>');
+    }
+    $('.delete').on('click', (e) => {
+      e.stopImmediatePropagation();
+      /* eslint-disable */
```
Comment (drteresavasquez) [inline src/javascripts/components/views/weaponsView.js]: Still am not liking this. What are we doing here? Let's set the ID to `e.currentTarget.id.split('__')[1]` if necessary. You are not using `type` anywhere that I see

**Review by drteresavasquez — CHANGES_REQUESTED** [review]
Some requested changes that will clean code

File: src/javascripts/components/views/weaponsView.js (hunk 3 above)
Comment (willkotheimer) [reply src/javascripts/components/views/weaponsView.js]: Fair! I'll take care of it!

File: src/javascripts/components/cards/weaponsCards.js (hunk 1 above)
Comment (willkotheimer) [reply src/javascripts/components/cards/weaponsCards.js:19]: Thanks!

Comment (willkotheimer) [reply src/javascripts/components/cards/weaponsCards.js:19]: Fixed!

File: src/javascripts/components/forms/updateWeaponsForm.js (hunk 2 above)
Comment (willkotheimer) [reply src/javascripts/components/forms/updateWeaponsForm.js]: Removed.

File: src/javascripts/components/views/weaponsView.js (hunk 3 above)
Comment (willkotheimer) [reply src/javascripts/components/views/weaponsView.js]: Done!

**Review by drteresavasquez — APPROVED** [review]

**Pull request merged**